src/sys/linux/hardwareinfo.rs

```diff
@@ -0,0 +1,38 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fs::File;
+use std::io::Read;
+use ::sys::linux::helpers;
+
+impl ::HardwareInfo for ::HostInfo {
+    fn online_cpu_count(&self) -> u32 {
+        let mut file = File::open("/sys/devices/system/cpu/online").unwrap();
```

Comment:
I think this should get revisited in the future and doesn't have to be solved in this commit.